cmd/orch-tester/orch_tester.go

```diff
@@ -470,75 +587,52 @@ func (s *streamerClient) getFinishedStats(mid string) (*streamerModel.Stats, err
 	return &stats, nil
 }
 
-func (s *streamerClient) avgSegDuration() (float64, error) {
-	val, err := s.queryVectorMetric(
-		"rate(livepeer_source_segment_duration_seconds_sum[1m])/rate(livepeer_source_segment_duration_seconds_count[1m])",
-	)
-	if err != nil {
-		return 0, err
+func avgMetric(streamer *streamerClient, embedded *broadcasterMetrics, metric string) float64 {
+	if embedded != nil {
+		return embedded.avg(metric)
 	}
-	valFloat := float64(0)
-	if val.Len() > 0 {
-		valFloat = float64((*val)[0].Value)
-		if math.IsNaN(valFloat) {
-			return 0, nil
-		}
-	}
-	return valFloat, nil
-}
 
-func (s *streamerClient) avgUploadTime() (float64, error) {
-	val, err := s.queryVectorMetric(
-		"rate(livepeer_upload_time_seconds_sum[1m])/rate(livepeer_upload_time_seconds_count[1m])",
-	)
+	query := fmt.Sprintf("sum(rate(livepeer_%s_sum[1m]))/sum(rate(livepeer_%s_count[1m]))", metric, metric)
```

Comment:
Hmm previously it didn't even have the `sum` right? I think it makes sense to have it, but any idea how it worked before? Is it because we always had a single instance of the orch-tester broadcaster anyway? (how does it not mix the metrics with regular non-orch-tester metircs?)

Reply:
Hmm, so the story is like this. We had 4 different metrics here:
- `transcode_overall_latency_seconds` => We already used the `sum`, so the query didn't change. The metric is exported with different tags, so we need to sum over different tags
- `upload_time_seconds` => We didn't use `sum`, but it was a bug [1]. The metric is exported with different tags, so we should sum it over different tags
- `source_segment_duration_seconds`, `download_time_seconds` => This metrics are not exported with tags, so technically we don't need to sum them; but `sum` does not harm and will work even if we introduce tags for these metrics at some point

[1] https://github.com/livepeer/stream-tester/issues/138